fuzzyfinder.go

```diff
@@ -725,6 +750,20 @@ func (f *finder) Find(slice interface{}, itemFunc func(i int) string, opts ...Op
 	return res[0], err
 }
 
+func (f *finder) Search(itemsFn func() []string, opts ...Option) (int, error) {
```

Comment:
I'm not certain if we should create a new export function.
Currently, this library provides `Find` and `FindMulti`, and `FindMulti` can select multiple items. So, if we create `Search` to select multiple items, also we need to create another export function for the same.

Personally, I think it would be simpler to keep the existing functions and ignore `itemFunc` argument.
What do you think about it?

Reply:
I struggled with this new export as well. But to be honest I don't see why it needed a `slice` and `itemFunc` in the first place. Would you mind explain a little bit?
It seems to me that it's much simpler to revamp the `Find` and `FindMulti` to accept a `func() []string` and remove the need for `reflect`. Of coz it breaks current api so maybe it's a `v2` thing.

Reply:
> It seems to me that it's much simpler to revamp the Find and FindMulti to accept a func() []string

I agree in part. As you said, it would be simpler to accept `func() []string` instead of a slice value in the aspect of the library developer. It eliminates reflection code and `interface{}` argument.

However, in the aspect of library users, I can't be sure that it's a better API. Since the new interface accepts items via `func() []string`, callers have to refill their entities to `[]string`.

For example, 

``` go
type Track struct {
    Name      string
    AlbumName string
    Artist    string
}
var tracks = []Track{
    {"foo", "album1", "artist1"},
    {"bar", "album1", "artist1"},
    {"foo", "album2", "artist1"},
    {"baz", "album2", "artist2"},
    {"baz", "album3", "artist2"},
}

idx, err := fuzzyfinder.Search(func() []string {
		items := make([]string, 0, len(tracks))
		for _, t := range tracks {
			items = append(items, t.Name)
		}

		return items
})
```

The current interface:

``` go 
idx, err := fuzzyfinder.Find(tracks, func(i int) string {
    return tracks[i].Name
})
```

I think the current interface is more developer-friendly than the interface of `Search`. By the way, this interface was inspired by `sort.Slice` of the standard package.

Reply:
It looks great to add `WithHotReloadLock(*sync.Mutex)`!
It looks like we can achieve what we want to do with a few changes to the existing usage. Can you go ahead with this approach?

> Btw in my experience it is rare to see function that asks for a pointer to slice, maybe we should mention that in the documentation.

We already mentioned about it in the documentation for [`WithHotReload`](https://pkg.go.dev/github.com/ktr0731/go-fuzzyfinder#WithHotReload) :)